common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -740,6 +746,61 @@ def associate_by_email_if_login_api(auth_entry, backend, details, user, current_
             return association_response
 
 
+@partial.partial
+def associate_by_email_if_saml(auth_entry, backend, details, user, strategy, *args, **kwargs):
+    """
+    This pipeline step associates the current social auth with the user with the
+    same email address in the database.  It defers to the social library's associate_by_email
+    implementation, which verifies that only a single database user is associated with the email.
+
+    This association is done ONLY if the user entered the pipeline belongs to SAML provider.
+    """
+    # pylint: disable=too-many-nested-blocks
```

Comment:
Is it necessary to disable this pylint check?  Seems like an easy refactor and would help on the readability/testability front.